tests/kernel_bundle/sycl_compile.h

```diff
@@ -0,0 +1,161 @@
+  auto dev = queue.get_device();
+
+  auto k_id = sycl::get_kernel_id<kernel_t>();
+  // Test can be skipped if all devices do not support online compilation
+  // since this is not sycl::compile's fault
+  if (!sycl::has_kernel_bundle<sycl::bundle_state::input>(ctx, {dev}, {k_id})) {
+    log.skip("No kernel bundle with input state with kernel: " + kName +
+             " (skipped).");
+    return;
+  }
+
+  auto input_kb =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(ctx, {dev}, {k_id});
+  auto input_ids = input_kb.get_kernel_ids();
+
+  auto obj_kb = compile_bundle<Overload>(input_kb, {dev});
+  kernel_bundle::define_kernel<DescriptorT, sycl::bundle_state::input>(queue);
+
+  auto restrictions{
+      kernel_bundle::get_restrictions<DescriptorT,
+                                      sycl::bundle_state::input>()};
+  bool dev_is_compat = restrictions.is_compatible(dev);
+
+  if (obj_kb.has_kernel(k_id) != dev_is_compat) {
+    FAIL(log, "Device does not support kernel " + kName +
+                  " but compiled bundle"
+                  " contains it");
+  }
+
+  // Check that result object bundle has the same kernels as input bundle
+  bool same_kernels_in_obj_bundle =
+      std::any_of(input_ids.cbegin(), input_ids.cend(),
+                  [&](auto &in_id) { return obj_kb.has_kernel(in_id); });
+
+  if (!same_kernels_in_obj_bundle) {
+    FAIL(log,
+         "Result bundle does not contain all kernels from input bundle "
+         "(kernel: " +
+             kName + ")");
+  }
+
+  // Check that input and result kernel bundles have the same context
+  if (input_kb.get_context() != obj_kb.get_context()) {
+    FAIL(log,
+         "Input bundle and result bundle have different contexts "
+         "(kernel: " +
+             kName + ")");
+  }
+}
+
+/** @brief Check that sycl::compile result bundle has the same associated
+ *         devices as presented in input device vector
+ *  @tparam Overload Selected overload of sycl::compile
+ */
+template <CompileOverload Overload>
+static void check_associated_devices(util::logger &log) {
+  auto queue = util::get_cts_object::queue();
+  auto ctx = queue.get_context();
+  std::vector<sycl::device> devices{queue.get_device()};
+  // Force duplicates
+  devices.push_back(devices[0]);
+
+  // Test can be skipped if all devices do not support online compilation
+  // since this is not sycl::compile's fault
+  if (!sycl::has_kernel_bundle<sycl::bundle_state::input>(ctx)) {
+    log.skip("No kernel bundle with input state for test (skipped).");
+    return;
+  }
+
+  auto input_kb =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(ctx, devices);
+  auto obj_kb = compile_bundle<Overload>(input_kb, devices);
+
+  // Check that result kernel bundle contains all devices from passed vector
+  bool same_devs = true;
+  auto kb_devs = obj_kb.get_devices();
+  same_devs = std::equal(kb_devs.begin(), kb_devs.end(), devices.begin());
+  if (!same_devs) {
```

Comment:
```suggestion
  auto kb_devs = obj_kb.get_devices();
  if (!same_devs = std::equal(kb_devs.begin(), kb_devs.end(), devices.begin())) {
```
and in C++20 it will be just 1 line :-)

Reply:
I've applied this comment [here](https://github.com/KhronosGroup/SYCL-CTS/pull/226/commits/269dcf45c5db46690f9568a15ec437558e76df3c#diff-d7dd04b2c33a3f23629a9f87a1e1089058ce7e6a7fa093db890c2d2508b4d206R134-R136)